src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/DeleteManagedOrganizationUserAccountCommand.cs

```diff
@@ -98,6 +114,7 @@ public async Task DeleteUserAsync(Guid organizationId, Guid organizationUserId,
             }
         }
 
+        await DeleteManyAsync(users);
```

Comment:
`DeleteManyAsync` returns a result object which is unused. It needs to add to the `results` variable we already have. Also, its validation logic should be better incorporated into the validation steps above.

I suggest the simplest way to do this is:
- refactor the private `ValidateDeleteUsersAsync` you added to take a single user. The method is just a loop anyway. Maybe rename it something to differentiate it, e.g. `ValidateUserMembershipAndPremium`
- also refactor it to throw its error message
- call it from within the loop above, just before `results.Add((orgUserId, string.Empty));`. That way any error will be caught and added to the existing results object
- then by the time `DeleteManyAsync` is called, we know all validation has passed